pkg/api/types.go

```diff
@@ -193,6 +193,126 @@ type VolumeSource struct {
+	// This is useful for development and testing only.
+	// on-host storage is not supported in any way
+	HostPath *HostPathVolumeSource `json:"hostPath"`
+}
+
+type PersistentVolume struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	//Spec defines a persistent volume owned by the cluster
+	Spec PersistentVolumeSpec `json:"spec,omitempty"`
+
+	// Status represents the current information about persistent volume.
+	Status PersistentVolumeStatus `json:"status,omitempty"`
+}
+
+type PersistentVolumeSpec struct {
+	// Resources represents the actual resources of the volume
+	Capacity ResourceList `json:"capacity`
+	// Source represents the location and type of a volume to mount.
+	// AccessModeTypes are inferred from the Source.
+	PersistentVolumeSource `json:",inline"`
+	// holds the binding reference to a PersistentVolumeClaim
+	ClaimRef *ObjectReference `json:"claimRef,omitempty"`
+}
+
+type PersistentVolumeStatus struct {
+	// Phase indicates if a volume is available, bound to a claim, or released by a claim
+	Phase PersistentVolumePhase `json:"phase,omitempty"`
+}
+
+type PersistentVolumeList struct {
+	TypeMeta `json:",inline"`
+	ListMeta `json:"metadata,omitempty"`
+	Items    []PersistentVolume `json:"items,omitempty"`
+}
+
+// a PersistentVolumeClaim is a user's request for and claim to a persistent volume
+type PersistentVolumeClaim struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the volume
+	Spec PersistentVolumeClaimSpec `json:"spec,omitempty"`
+
+	// Status represents the current information about a persistent volume.
+	Status PersistentVolumeClaimStatus `json:"status,omitempty"`
+}
+
+type PersistentVolumeClaimList struct {
+	TypeMeta `json:",inline"`
+	ListMeta `json:"metadata,omitempty"`
+	Items    []PersistentVolumeClaim `json:"items,omitempty"`
+}
+
+// a PersistentVolumeClaimSpec describes the common attributes of storage devices
+// and allows a Source for provider-specific attributes
+type PersistentVolumeClaimSpec struct {
+	// Contains the types of access modes required
+	AccessModes []AccessModeType `json:"accessModes,omitempty"`
+	// Resources represents the minimum resources required
+	Resources ResourceRequirements `json:"resources,omitempty"`
+}
+
+type PersistentVolumeClaimStatus struct {
+	// Phase represents the current phase of PersistentVolumeClaim
+	Phase PersistentVolumeClaimPhase `json:"phase,omitempty"`
+	// AccessModes contains all ways the volume backing the PVC can be mounted
+	AccessModes []AccessModeType `json:"accessModes,omitempty`
+	// Represents the actual resources of the underlying volume
+	Capacity ResourceList `json:"capacity,omitempty"`
+	// VolumeRef is a reference to the PersistentVolume bound to the PersistentVolumeClaim
+	VolumeRef *ObjectReference `json:"volumeRef,omitempty"`
+}
+
+type AccessModeType string
+
+const (
+	ReadWriteOnce AccessModeType = "ReadWriteOnce"
```

Comment:
Yeah, please retartet feedback to #5800 - that's the one I want to commit
ASAP

On Mon, Mar 23, 2015 at 11:33 AM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5105#discussion_r26965220>
> :
>
> > +
> > +type PersistentVolumeClaimStatus struct {
> > +	// Phase represents the current phase of PersistentVolumeClaim
> > +	Phase PersistentVolumeClaimPhase `json:"phase,omitempty"`
> > +	// AccessModes contains all ways the volume backing the PVC can be mounted
> > +	AccessModes []AccessModeType `json:"accessModes,omitempty`
> > +	// Represents the actual resources of the underlying volume
> > +	Capacity ResourceList `json:"capacity,omitempty"`
> > +	// VolumeRef is a reference to the PersistentVolume bound to the PersistentVolumeClaim
> > +	VolumeRef *ObjectReference `json:"volumeRef,omitempty"`
> > +}
> > +
> > +type AccessModeType string
> > +
> > +const (
> > +	ReadWriteOnce AccessModeType = "ReadWriteOnce"
>
> @quinton-hoole <https://github.com/quinton-hoole> agreed. I added
> comments to each constant in #5800
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5800>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5105/files#r26965220>
> .
>